src/core/locale.ts

```diff
@@ -37,7 +37,7 @@ const localeModels: Dictionary<Model> = {};
 export const SYSTEM_LANG = !env.domSupported ? DEFAULT_LOCALE : (function () {
     const langStr = (
         /* eslint-disable-next-line */
-        document.documentElement.lang || navigator.language || (navigator as any).browserLanguage
+        document.documentElement.lang || navigator.language || (navigator as any).browserLanguage || DEFAULT_LOCALE
```

Comment:
Browser: Firefox ESR 52.5.2_x86

<img width="716" alt="image" src="https://github.com/apache/echarts/assets/26894991/b553df15-8014-474d-bd1f-79cd35e95984">

Reply:
Though I tested it in Firefox ESR 52.5.2 (Win64) and found the language is not undefined, adding a fallback to prevent potential errors is always good.

![image](https://github.com/apache/echarts/assets/26999792/0084f77f-fa49-4b5b-a449-a7e8aff3df59)

Reply:
Yes, It can meet some customized terminals (customized browser).